src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -189,11 +194,18 @@ export class CdkVirtualScrollViewport extends CdkScrollable implements OnInit, O
     this._viewportChanges = viewportRuler.change().subscribe(() => {
       this.checkViewportSize();
     });
+
+    if (!this.scrollable) {
+      // No scrollable is provided, so the virtual-scroll-viewport needs to become a scrollable
+      renderer.addClass(this.elementRef.nativeElement, 'cdk-virtual-scrollable');
```

Comment:
nit: we don't use `Renderer2`, you can just do `...nativeElement.classList.add(...)`

Reply:
done. Could you provide some Information or links, why not using `Renderer2`?  I guess I saw it being used somewhere else in angular code.

Reply:
IIRC `Renderer2` was originally a consideration for SSR, but these days SSR uses Domino to support native DOM APIs in SSR. We don't mention anything about this in our [coding practices](https://github.com/angular/components/blob/master/CODING_STANDARDS.md), but we probably should add a note.